### PR TITLE
Create a QbftMetricsService

### PR DESCRIPTION
**Pull request by rain-on**
## PR description
Adds the query service so that the a plugin can query how the QBFT validator set are performing.

## Fixed Issue(s)
N/A

## Changelog
- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/BftQueryService.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import java.util.Collection;
 
 /** Allows for the IBFT 2.0 specific aspects of the block chain to be queried. */
-public interface IbftQueryService extends PoaQueryService {
```
Comment (jframe) [inline plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/BftQueryService.java:23]: We should leave the IbftQueryService as is and mark it deprecated and add the BftQueryService. Perhaps the PoAMetricsService can now finally be removed now too.

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.plugin.data.Address;
+import org.hyperledger.besu.plugin.services.query.BftQueryService;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.apache.tuweni.bytes.Bytes32;
+
+public class QbftQueryServiceImpl extends PoaQueryServiceImpl implements BftQueryService {
```
Comment (jframe) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java:33]: Can you add a test for this too

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  private BlockHeader getHeaderFromChain(
+      final org.hyperledger.besu.plugin.data.BlockHeader header) {
+    if (header instanceof BlockHeader) {
+      return (BlockHeader) header;
+    }
+
+    final Hash blockHash = Hash.wrap(Bytes32.wrap(header.getBlockHash().toArray()));
+    return getBlockchain().getBlockHeader(blockHash).orElseThrow();
+  }
```
Comment (jframe) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java:65]: I know the code is simple, but duplicating the ibft query service code isn't nice. We could just reuse the existing ibft query service renaming it in the process. If the headers change we can split the class.

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/IbftQueryService.java [hunk 4]
```diff
@@ -19,7 +19,15 @@
 
 import java.util.Collection;
 
-/** Allows for the IBFT 2.0 specific aspects of the block chain to be queried. */
+/**
+ * Allows for the IBFT 2.0 specific aspects of the block chain to be queried.
+ *
+ * <p>* @deprecated This interface has been replaced by {@link *
```
Comment (jframe) [inline plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/IbftQueryService.java:25]: This comment doesn't seem quite right. I think what you have about the BftQueryService is what should go here.

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.plugin.data.Address;
+import org.hyperledger.besu.plugin.services.query.BftQueryService;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.apache.tuweni.bytes.Bytes32;
+
+public class QbftQueryServiceImpl extends PoaQueryServiceImpl implements BftQueryService {
```
Comment (jframe) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java:33]: Is this still needed? I thought this was replaced by the BftQueryServiceImpl

Comment (rain-on) [reply consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/queries/QbftQueryServiceImpl.java:33]: true-  should be removed. ta.

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/BftQueryService.java (hunk 1 above)
Comment (rain-on) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/BftQueryService.java:23]: Flagged as deprecated, but would prefer to remove the PoAMetricsService in a separate PR

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/IbftQueryService.java (hunk 4 above)
Comment (rain-on) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/IbftQueryService.java:25]: links to the BFTQueryService now (is that what you had in mind?)

**Review by jframe — APPROVED** [review]

**Pull request merged**